app/models/concerns/due_date_actions.rb

```diff
@@ -0,0 +1,300 @@
+    when 'metareview'
+      metareview_permissible?
+    when 'quiz'
+      quiz_permissible?
+    when 'team_formation'
+      team_formation_permissible?
+    when 'signup'
+      signup_permissible?
+    when 'drop_topic'
+      drop_topic_permissible?
+    else
+      false
+    end
+  end
+
+  # Get all currently allowed actions
+  def allowed_actions
+    actions = []
+    actions << 'submission' if submission_permissible?
+    actions << 'review' if review_permissible?
+    actions << 'teammate_review' if teammate_review_permissible?
+    actions << 'metareview' if metareview_permissible?
+    actions << 'quiz' if quiz_permissible?
+    actions << 'team_formation' if team_formation_permissible?
+    actions << 'signup' if signup_permissible?
+    actions << 'drop_topic' if drop_topic_permissible?
+    actions
+  end
+
+  # Check if any actions are currently allowed
+  def has_allowed_actions?
+    allowed_actions.any?
+  end
+
+  # Get permission summary for all actions
+  def action_permissions_summary
+    {
+      submission: submission_permissible?,
+      review: review_permissible?,
+      teammate_review: teammate_review_permissible?,
+      metareview: metareview_permissible?,
+      quiz: quiz_permissible?,
+      team_formation: team_formation_permissible?,
+      signup: signup_permissible?,
+      drop_topic: drop_topic_permissible?,
+      has_any_permissions: has_allowed_actions?
+    }
+  end
+
+  # Topic-specific permission checking
+  def activity_permissible_for_topic?(activity, topic_id)
+    deadline = current_stage_for_topic(topic_id, activity)
+    return false unless deadline
+
+    deadline.activity_permissible?(activity)
+  end
+
+  # Topic-specific syntactic sugar methods
+  def submission_permissible_for_topic?(topic_id)
+    activity_permissible_for_topic?(:submission, topic_id)
+  end
+
+  def review_permissible_for_topic?(topic_id)
+    activity_permissible_for_topic?(:review, topic_id)
+  end
+
+  def quiz_permissible_for_topic?(topic_id)
+    activity_permissible_for_topic?(:quiz, topic_id)
+  end
+
+  # Copy all due dates to a new parent object
+  def copy_due_dates_to(new_parent)
+    due_dates.find_each do |due_date|
+      due_date.copy_to(new_parent)
+    end
+  end
+
+  # Duplicate due dates with modifications
+  def duplicate_due_dates_with_changes(new_parent, changes = {})
+    due_dates.map do |due_date|
+      due_date.duplicate_with_changes(changes.merge(parent: new_parent))
+    end
+  end
+
+  # Create a new due date for this parent
+  def create_due_date(deadline_type_name, due_at, round: 1, **attributes)
+    deadline_type = DeadlineType.find_by_name(deadline_type_name)
+    raise ArgumentError, "Invalid deadline type: #{deadline_type_name}" unless deadline_type
+
+    due_dates.create!(
+      deadline_type: deadline_type,
+      due_at: due_at,
+      round: round,
+      **attributes
+    )
+  end
+
+  # Update or create a due date for a specific type and round
```

Comment:
I am confused about where this method would be needed.  Is this the way you would change permissions? e.g., change review_allowed_id from "OK" to "no"?  Wouldn't it be clearer to have simple getters and setters for the different fields in due_date?